src/utils/index.ts

```diff
@@ -163,3 +163,37 @@ export function assertIsTruthy<T>(val: any): asserts val is NonNullable<T> {
     throw new AssertionError({ expected: true, actual: val })
   }
 }
+
+export enum IdentityNameValidityError {
+  MINIMUM_LENGTH = 'error_minimum_length',
+  MAXIMUM_LENGTH = 'error_maximum_length',
+  ILLEGAL_CHARACTER = 'error_illegal_character',
+  UNAVAILABLE = 'error_name_unavailable'
+}
+
+const containsLegalCharacters = (name: string) => /^[a-z0-9_]+$/.test(name)
+
+export const validateIdentityName = (identityName: string): IdentityNameValidityError | null => {
+  const nameLength = identityName.length
+
+  if (nameLength < 8) {
+    return IdentityNameValidityError.MINIMUM_LENGTH
+  }
+
+  if (nameLength > 38) {
+    return IdentityNameValidityError.MINIMUM_LENGTH
+  }
+
+  if (!containsLegalCharacters(identityName)) {
+    return IdentityNameValidityError.ILLEGAL_CHARACTER
+  }
+
+  return null
+}
+
+export const checkIdentityNameAvailability = async (name: string, subdomain = 'id.blockstack') => {
+  const url = `https://registrar.blockstack.org/v1/names/${name.toLowerCase()}.${subdomain}`
```

Comment:
This will only work for the `.id.blockstack` registrar. We need to have `subdomain` be the `Subdomain` enum, which will also let you get the registrar URL in a type-safe manner.

I also think you don't need the full subdomain at the end of this URL - only `name`

Reply:
Didn't seem to work without subdomain, moved to using enum